include/ht_material.h

```diff
@@ -58,10 +58,16 @@ namespace Hatchit {
             virtual bool VPrepare(IPipeline* pipeline) = 0;
             virtual bool VUpdate() = 0;
 
-        protected:
+            inline IPipeline* GetPipeline() 
```

Comment:
No implementation in interfaces.  If need be, have a PipelineBase abstract class.